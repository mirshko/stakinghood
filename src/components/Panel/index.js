import React from "react";
import { Box, Card } from "rebass";
import PropTypes from "prop-types";

const PseudoPanel = ({ height }) => (
  <Card borderRadius={4} bg="white" m={10} style={{ height: height }} />
);

const Panel = ({ children }) => (
  <Card
    boxShadow={[
      "none",
      `rgba(94, 111, 243, 0.16) 0px 48px 80px,
  rgba(22, 25, 49, 0.16) 0px 16px 32px, rgba(22, 25, 49, 0.08) 0px 6px 16px`
    ]}
    borderRadius={[0, 24]}
    bg="#F5F6F9"
    mt={[0, 32]}
    mb={[0, 128]}
    style={{ overflow: "hidden" }}
  >
    <Box px={20} pt={20} pb={10} mb={10} bg="white">
      {children}
    </Box>
    <PseudoPanel height={40} />
    <PseudoPanel height={96} />
  </Card>
);

Panel.propTypes = {
  children: PropTypes.node.isRequired
};

export default Panel;
