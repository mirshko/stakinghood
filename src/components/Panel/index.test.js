import React from "react";
import { shallow } from "enzyme";

import Panel from "./index";

describe("<Panel />", () => {
  it("it works", () => {
    const wrapper = shallow(
      <Panel>
        <p>Children</p>
      </Panel>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
