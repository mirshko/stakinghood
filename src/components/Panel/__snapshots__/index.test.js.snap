// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> it works 1`] = `
<Styled(styled.div)
  bg="#F5F6F9"
  borderRadius={
    Array [
      0,
      24,
    ]
  }
  boxShadow={
    Array [
      "none",
      "rgba(94, 111, 243, 0.16) 0px 48px 80px,
  rgba(22, 25, 49, 0.16) 0px 16px 32px, rgba(22, 25, 49, 0.08) 0px 6px 16px",
    ]
  }
  mb={
    Array [
      0,
      128,
    ]
  }
  mt={
    Array [
      0,
      32,
    ]
  }
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <styled.div
    bg="white"
    mb={10}
    pb={10}
    pt={20}
    px={20}
  >
    <p>
      Children
    </p>
  </styled.div>
  <PseudoPanel
    height={40}
  />
  <PseudoPanel
    height={96}
  />
</Styled(styled.div)>
`;
